mq/plugins/broFixup.py

```diff
@@ -41,39 +45,493 @@ def __init__(self):
 
         self.registration = ['bro', 'nsm']
         self.priority = 5
+        try:
+            self.mozdefhostname = u'{0}'.format(node())
+        except:
+            self.mozdefhostname = 'failed to fetch mozdefhostname'
+            pass
+
 
     def onMessage(self, message, metadata):
         # set the doc type to bro
         # to avoid data type conflicts with other doc types
         # (int v string, etc)
-        metadata['doc_type']= 'bro'
+        # index holds documents of type 'type'
+        # index -> type -> doc
+        metadata['doc_type']= 'nsm'
+        whitelist = ['hostname', 'type', 'tags', 'customendpoint']
+
+        # move Bro specific fields under 'details' while preserving metadata
+        newmessage = dict()
+
+        newmessage['details'] = message
+        newmessage['customendpoint'] = 'bro'
+
+
+        # add mandatory fields
+        if 'ts' in message.keys():
+            newmessage[u'utctimestamp'] = toUTC(message['ts']).isoformat()
+            newmessage[u'timestamp'] = toUTC(message['ts']).isoformat()
+        else:
+            # a malformed message somehow managed to crawl to us, let's put it somewhat together
+            newmessage[u'utctimestamp'] = toUTC(datetime.now()).isoformat()
+            newmessage[u'timestamp'] = toUTC(datetime.now()).isoformat()
+
+        newmessage[u'receivedtimestamp'] = toUTC(datetime.now()).isoformat()
+
+
+        # add mandatory fields
+        # move bro data under details
+        newmessage[u'eventsource'] = u'nsm'
+        newmessage[u'severity'] = u'INFO'
+        #newmessage[u'processname'] = u''
```

Comment:
If we want to keep these commented out, can we remove them to avoid any confusion?

Reply:
Removed